mpd/base.py

```diff
@@ -59,7 +61,18 @@ class ProtocolError(MPDError):
 
 
 class CommandError(MPDError):
-    pass
+    def __init__(self, error):
+        self.errno = None
+        self.offset = None
+        self.command = None
+        self.msg = None
+
+        match = ERROR_PATTERN.match(error)
+        if match:
+            self.errno = int(match.group("errno"))
```

Comment:
I assume you make this of type Errors?

Reply:
Ah yeah, importing ProtocolError from base.py yields circular imports. Awesome!